### PR TITLE
Metatable similar to lua.

### PR DESCRIPTION
**Pull request by st0le**

### CODE REVIEW

File: metatable.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        else:
+            ret = self[key] = self.default_factory(self,key)
+            return ret
+
+def fib(d,n):
+	if n == 0 or n == 1: return n
+	return d[n - 1] + d[n - 2]
+
+d = metatable(fib)
+print d[1]
```
Comment (brennerm) [inline metatable.py]: Please use print()

File: metatable.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+print d[3]
```
Comment (brennerm) [inline metatable.py]: Please use print()

File: metatable.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+print d[10]
```
Comment (brennerm) [inline metatable.py]: Please use print()

File: metatable.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+print d
```
Comment (brennerm) [inline metatable.py]: Please use print()

**Comment by brennerm** [comment]
Be sure to add yourself to the contributors :)

**Comment by st0le** [comment]
Will do.

**Pull request merged**